docs/howtos/vap-migration.md

```diff
@@ -0,0 +1,161 @@
+
+<details>
+
+<summary>`ValidatingAdmissionPolicy` definition - `vap.yml`</summary>
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "force-liveness-probe"
+spec:
+  failurePolicy: Fail
+  variables:
+    - name: containers_without_liveness_probe
+      expression: |
+        object.spec.template.spec.containers.filter(c, !has(c.livenessProbe)).map(c, c.name)
+  matchConstraints:
+    resourceRules:
+      - apiGroups: ["apps"]
+        apiVersions: ["v1"]
+        operations: ["CREATE", "UPDATE"]
+        resources: ["deployments"]
+  validations:
+    - expression: |
+        size(variables.containers_without_liveness_probe) == 0
+      messageExpression: |
+        'These containers are missing a liveness probe: ' + variables.containers_without_liveness_probe.join(' ')
+      reason: Invalid
+```
+
+</details>
+
+And a file containing the `ValidatingAdmissionPolicyBinding` resource:
+
+<details>
+
+<summary>`ValidatingAdmissionPolicyBinding` definition - `vap-binding.yml`</summary>
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1
+kind: ValidatingAdmissionPolicyBinding
+metadata:
+  name: "kw-scaffold-demo"
+spec:
+  policyName: "force-liveness-probe"
+  validationActions: [Deny]
+  matchResources:
+    namespaceSelector:
+      matchLabels:
+        docs.kubewarden.io/vap-migration: enabled
+```
+
+</details>
+
+You can migrate the policy to Kubewarden by following these steps:
+
+<details>
+
+<summary>`kwctl` command for policy migration</summary>
+
+```shell
+$ kwctl scaffold vap \
+    -p vap.yml \
+    -b vap-binding.yml
+```
+
+</details>
+
+The command produces output similar to this:
+
+<details>
+
+<summary>Output from the migration</summary>
+
+```shell
+2024-06-24T16:00:16.516062Z  WARN kwctl::scaffold: Using the 'latest' version of the CEL policy could lead to unexpected behavior. It is recommended to use a specific version to avoid breaking changes.
+apiVersion: policies.kubewarden.io/v1
+kind: ClusterAdmissionPolicy
+metadata:
+  name: kw-scaffold-demo
+spec:
+  module: ghcr.io/kubewarden/policies/cel-policy:latest
+  settings:
+    variables:
+    - expression: |
+        object.spec.template.spec.containers.filter(c, !has(c.livenessProbe)).map(c, c.name)
+      name: containers_without_liveness_probe
+    validations:
+    - expression: |
+        size(variables.containers_without_liveness_probe) == 0
+      messageExpression: |
+        'These containers are missing a liveness probe: ' + variables.containers_without_liveness_probe.join(' ')
+      reason: Invalid
+  rules:
+  - apiGroups:
+    - apps
+    apiVersions:
+    - v1
+    resources:
+    - deployments
+    operations:
+    - CREATE
+    - UPDATE
+  mutating: false
+  failurePolicy: Fail
+  namespaceSelector:
+    matchLabels:
+      docs.kubewarden.io/vap-migration: enabled
+```
+
+:::note
+The command complains about the usage of the `latest` version of the CEL policy.
```

Comment:
maybe "warns the user" instead of complains?

Reply:
good point, changed